### ISSUE TITLE
Freelook sensitivity setting brakes editor movement

### ISSUE THREAD

**Issue by ngreve** [godotengine/godot#65647]
### Godot version

v4.0.alpha.custom_build [d0a2a4c98]

### System information

Linux

### Issue description

When setting the `Freelook Sensitivity` to a lower value than the default value of 0.25 the camera starts lagging/gets stuck when looking around. The lower the value the more extreme the effect gets.

https://user-images.githubusercontent.com/18660100/189519869-73b5e61b-ef0b-4ac9-8005-3cbbe02370ca.mp4

### Steps to reproduce

1. Create a new project
2. Go to `Editor`->`Editor Settings`->`Editors`->`Freelook Sensitivity` set it to a lower value than 0.25. The lower the value the more extreme the effect
3. Try to look around in the editor

### Minimal reproduction project

Create a new one

**Comment by Calinou** [comment]
Does it improve if you increase freelook inertia in the editor settings?

**Comment by ngreve** [comment]
Unfortunately not. Playing around with the `Freelook Navigation Scheme`, `Freelook Base Speed` or enabling `Freelook Speed Zoom Link` also does not improve anything.

**Comment by ngreve** [comment]
TLDR; Quickfix: Deactivate `Editor`->`Editor Settings`->`Editors`->`3D`->`Warped Mouse Panning`

I narrowed down the cause of the problem, but I don't now how to fix it.

The root cause of the problem is a false value of the `p_relative` argument of the [Node3DEditorViewport::_nav_look()](https://github.com/godotengine/godot/blob/f6b36f54e50cff518b4085cf665405ad1b5f6d44/editor/plugins/node_3d_editor_plugin.cpp#L2168) function under certain circumstances.

It seems to me, that if you move your mouse fast enough that the mouse cursor would be outside of the viewport  in the next frame, the signs of `p_relative` toggles. So, the mouse movement is misinterpreted.  

Iv'e tested it,  and it turns out the lagging occurs **significantly** less often, the wider the godot window/viewport is, because with a wide viewport, the mouse cursor is less likely to be outside of it in the next frame.

If you [take a look at this line](https://github.com/godotengine/godot/blob/f6b36f54e50cff518b4085cf665405ad1b5f6d44/editor/plugins/node_3d_editor_plugin.cpp#L1818), you will see that the the `p_relative` argument is  generated by the [`Node3DEditorViewport::_get_warped_mouse_motion()`](https://github.com/godotengine/godot/blob/f6b36f54e50cff518b4085cf665405ad1b5f6d44/editor/plugins/node_3d_editor_plugin.cpp#L2290) function, which checks if mouse panning is activated.

If you turn of `Warped Mouse Panning` the issue disappears. (`Editor`->`Editor Settings`->`Editors`->`3D`->`Warped Mouse Panning`)

I hope my explanation is not too confusing. 

Anyway, can anyone tell me what the point of `Warped Mouse Panning` is?

**Comment by Calinou** [comment]
> Anyway, can anyone tell me what the point of Warped Mouse Panning is?

It moves the mouse cursor to the other end of the viewport when you pan using the middle mouse button and the mouse ends up outside the window.

**Comment by WhalesState** [comment]
Can't reproduce this in current master.
